paasta_tools/cleanup_marathon_jobs.py

```diff
@@ -118,45 +131,67 @@ def delete_app(app_id, client, soa_dir):
         raise
 
 
-def cleanup_apps(soa_dir):
+def cleanup_apps(soa_dir, kill_threshold=0.5, force=False):
     """Clean up old or invalid jobs/apps from marathon. Retrieves
     both a list of apps currently in marathon and a list of valid
     app ids in order to determine what to kill.
 
-    :param soa_dir: The SOA config directory to read from"""
+    :param soa_dir: The SOA config directory to read from
+    :param kill_threshold: The decimal fraction of apps we think is
+        sane to kill when this job runs.
+    :param force: Force the cleanup if we are above the kill_threshold"""
     log.info("Loading marathon configuration")
     marathon_config = marathon_tools.load_marathon_config()
     log.info("Connecting to marathon")
     client = marathon_tools.get_marathon_client(marathon_config.get_url(), marathon_config.get_username(),
                                                 marathon_config.get_password())
 
-    valid_services = get_services_for_cluster(instance_type='marathon', soa_dir=soa_dir)
+    valid_apps = get_services_for_cluster(instance_type='marathon', soa_dir=soa_dir)
     running_app_ids = marathon_tools.list_all_marathon_app_ids(client)
 
+    running_apps = []
     for app_id in running_app_ids:
-        log.debug("Checking app id %s", app_id)
         try:
-            service, instance, _, __ = marathon_tools.deformat_job_id(app_id)
+            app_id = marathon_tools.deformat_job_id(app_id)
         except InvalidJobNameError:
             log.warn("%s doesn't conform to paasta naming conventions? Skipping." % app_id)
             continue
-        if (service, instance) not in valid_services:
-            delete_app(
-                app_id=app_id,
-                client=client,
-                soa_dir=soa_dir,
-            )
+        running_apps.append(app_id)
+    apps_to_kill = [running_app for running_app in running_apps
+                    if running_app[0:2] not in valid_apps]
```

Comment:
I also felt that the `[0:2]` was a bit nasty, that version is much more readable 👍